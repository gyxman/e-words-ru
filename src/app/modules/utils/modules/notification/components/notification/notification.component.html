<div *ngIf="info" class="wrapper" [ngClass]="classes">
    <div class="icon-wrapper">
        <mat-icon
            automation-id="icon"
            class="icon"
            aria-hidden="true"
            [svgIcon]="info.type"
        ></mat-icon>
    </div>
    <div class="info" [innerHTML]="info.text"></div>
</div>
