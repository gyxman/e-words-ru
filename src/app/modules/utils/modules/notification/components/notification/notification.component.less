@import '../../../../../../../styles/index';

:host {
    display: block;
    margin: 8px;
}

.wrapper {
    display: flex;
    max-height: 60px;
    box-shadow: 0 2px 8px 0 fade(@color-black, 24%);
    overflow: hidden;

    &_success {
        & .icon-wrapper {
            background-color: @color-green-light;
        }
    }

    &_warning {
        & .icon-wrapper {
            background-color: @color-orange;
        }
    }

    &_error {
        & .icon-wrapper {
            background-color: @color-red;
        }
    }
}

.icon-wrapper {
    position: relative;
    display: flex;
    align-items: center;
    padding-left: 15px;
    margin-right: 30px;
    flex-shrink: 0;
    color: @color-white;

    &:before {
        .clear();
        position: absolute;
        top: 50%;
        right: -20px;
        width: 50px;
        height: 50px;
        background-color: inherit;
        transform: rotate(45deg) translateY(-50%);
        transform-origin: top;
    }
}

.icon {
    position: relative;
    z-index: 2;
    width: 24px;
    height: 24px;
}

.info {
    .font-p2();
    padding: 10px;
}
