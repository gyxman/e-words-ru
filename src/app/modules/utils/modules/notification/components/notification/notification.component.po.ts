import {PageObject} from '../../../../../../testing/page-object';
import {DebugElement} from '@angular/core';

export class NotificationComponentPo<T> extends PageObject<T> {
    get icon(): DebugElement {
        return this.getByAutomationId('icon');
    }
}
