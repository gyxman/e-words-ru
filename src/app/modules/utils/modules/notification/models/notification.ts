export type NotificationModel = Readonly<{
    text: string;
    type: 'success' | 'warning' | 'error';
    time?: number;
}>;
