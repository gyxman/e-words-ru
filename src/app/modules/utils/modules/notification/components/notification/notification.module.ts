import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {NotificationComponent} from './notification.component';
import {MatIconModule} from '@angular/material/icon';

@NgModule({
    imports: [CommonModule, MatIconModule],
    declarations: [NotificationComponent],
    exports: [NotificationComponent],
})
export class NotificationModule {}
