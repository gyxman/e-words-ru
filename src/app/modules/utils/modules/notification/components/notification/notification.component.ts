import {ChangeDetectionStrategy, Component, HostBinding, Input} from '@angular/core';
import {NotificationModel} from '../../models/notification';
import {animate, state, style, transition, trigger} from '@angular/animations';
import {MatIconRegistry} from '@angular/material/icon';
import {DomSanitizer} from '@angular/platform-browser';

@Component({
    selector: 'app-notification',
    templateUrl: './notification.component.html',
    styleUrls: ['./notification.component.less'],
    changeDetection: ChangeDetectionStrategy.OnPush,
    animations: [
        trigger('expand', [
            state('in', style({height: '*'})),
            transition(':leave', [
                style({height: '*'}),
                animate(
                    200,
                    style({height: 0, opacity: 0, transform: 'translateX(110%)'}),
                ),
            ]),
            transition(':enter', [
                style({height: 0, opacity: 0, transform: 'translateX(110%)'}),
                animate(
                    200,
                    style({height: '*', opacity: 1, transform: 'translateX(0)'}),
                ),
            ]),
        ]),
    ],
})
export class NotificationComponent {
    @HostBinding('@expand') readonly expand;

    @Input() info: NotificationModel;

    constructor(iconRegistry: MatIconRegistry, sanitizer: DomSanitizer) {
        iconRegistry.addSvgIcon(
            'success',
            sanitizer.bypassSecurityTrustResourceUrl(
                'assets/icons/notifications/icon-success.svg',
            ),
        );
        iconRegistry.addSvgIcon(
            'warning',
            sanitizer.bypassSecurityTrustResourceUrl(
                'assets/icons/notifications/icon-warning.svg',
            ),
        );
        iconRegistry.addSvgIcon(
            'error',
            sanitizer.bypassSecurityTrustResourceUrl(
                'assets/icons/notifications/icon-error.svg',
            ),
        );
    }

    get classes() {
        const {type} = this.info;

        return {
            [`wrapper_${type}`]: true,
        };
    }
}
