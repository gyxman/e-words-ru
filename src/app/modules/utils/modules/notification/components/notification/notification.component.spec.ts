import {async, ComponentFixture, TestBed} from '@angular/core/testing';

import {NotificationComponent} from './notification.component';
import {NoopAnimationsModule} from '@angular/platform-browser/animations';

describe('NotificationComponent', () => {
    let component: NotificationComponent;
    let fixture: ComponentFixture<NotificationComponent>;

    beforeEach(async(() => {
        TestBed.configureTestingModule({
            imports: [NoopAnimationsModule],
            declarations: [NotificationComponent],
        }).compileComponents();
    }));

    beforeEach(() => {
        fixture = TestBed.createComponent(NotificationComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
    });

    it('should create', () => {
        expect(component).toBeTruthy();
    });
});
