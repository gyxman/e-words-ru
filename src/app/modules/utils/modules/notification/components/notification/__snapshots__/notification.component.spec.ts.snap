// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationComponent - компонент нотификации Если пользователь показывает нотификацию ошибки, то верстка корректна 1`] = `
<div
  id="root4"
>
  <app-notification
    class="ng-trigger ng-trigger-expand ng-animating"
    style=""
  >
    
    <div
      class="wrapper wrapper_error"
      style=""
    >
      <div
        class="icon-wrapper"
      >
        <mat-icon
          aria-hidden="true"
          automation-id="icon"
          class="icon"
        />
      </div>
      <div
        class="info"
      >
        Нотификация ошибки
      </div>
    </div>
  </app-notification>
</div>
`;

exports[`NotificationComponent - компонент нотификации Если пользователь показывает нотификацию уведомления, то верстка корректна 1`] = `
<div
  id="root2"
>
  <app-notification
    class="ng-trigger ng-trigger-expand ng-animating"
    style=""
  >
    
    <div
      class="wrapper wrapper_warning"
      style=""
    >
      <div
        class="icon-wrapper"
      >
        <mat-icon
          aria-hidden="true"
          automation-id="icon"
          class="icon"
        />
      </div>
      <div
        class="info"
      >
        Нотификация уведомления
      </div>
    </div>
  </app-notification>
</div>
`;

exports[`NotificationComponent - компонент нотификации Если пользователь показывает нотификацию успеха, то верстка корректна 1`] = `
<div
  id="root0"
>
  <app-notification
    class="ng-trigger ng-trigger-expand ng-animating"
    style=""
  >
    
    <div
      class="wrapper wrapper_success"
      style=""
    >
      <div
        class="icon-wrapper"
      >
        <mat-icon
          aria-hidden="true"
          automation-id="icon"
          class="icon"
        />
      </div>
      <div
        class="info"
      >
        Нотификация успеха
      </div>
    </div>
  </app-notification>
</div>
`;
